src/include/timestamp.i

```diff
@@ -0,0 +1,196 @@
+    dest->stop_txn = source->stop_txn;
+    dest->prepare = source->prepare;
+}
+
+/*
+ * __wt_time_window_is_empty --
+ *     Return true if the time window is equivalent to the default time window.
+ */
+static inline bool
+__wt_time_window_is_empty(WT_TIME_WINDOW *tw)
+{
+    if (tw->start_durable_ts == WT_TS_NONE && tw->start_ts == WT_TS_NONE &&
+      tw->start_txn == WT_TXN_NONE && tw->stop_durable_ts == WT_TS_NONE &&
+      tw->stop_ts == WT_TS_MAX && tw->stop_txn == WT_TXN_MAX)
+        return (true);
+    return (false);
+}
+
+/*
+ * __wt_time_windows_equal --
+ *     Return true if the time windows are the same.
+ */
+static inline bool
+__wt_time_windows_equal(WT_TIME_WINDOW *tw1, WT_TIME_WINDOW *tw2)
+{
+    if (tw1 > start_durable_ts == tw2 > start_durable_ts &&
+      tw1 > stop_durable_ts == tw2 > stop_durable_ts && tw1 > start_ts == tw2 > start_ts &&
+      tw1 > start_txn == tw2 > start_txn && tw1 > stop_ts == tw2 > stop_ts &&
+      tw1 > stop_txn == tw2 > stop_txn && tw1 > prepare == tw2 > prepare)
+        return (true);
+    return (false);
+}
+
+/*
+ * __wt_time_window_set_start --
+ *     Set the start values of a time window from those in an update structure.
+ */
+static inline void
+__wt_time_window_set_start(WT_TIME_WINDOW *tw, WT_UPDATE *upd)
+{
+    tw->start_ts = upd->start_ts;
+    tw->start_durable_ts = upd->durable_ts;
+    tw->start_txn = upd->txnid;
+}
+
+/*
+ * __wt_time_window_set_stop --
+ *     Set the start values of a time window from those in an update structure.
+ */
+static inline void
+__wt_time_window_set_stop(WT_TIME_WINDOW *tw, WT_UPDATE *upd)
+{
+    tw->stop_ts = upd->start_ts;
+    tw->stop_durable_ts = upd->durable_ts;
+    tw->stop_txn = upd->txnid;
+}
+
+/*
+ * __wt_time_aggregate_init --
+ *     Initialize the fields in an aggregated time window to their defaults.
+ */
+static inline void
+__wt_time_aggregate_init(WT_TIME_AGGREGATE *ta)
+{
+    ta->newest_start_durable_ts = WT_TS_NONE;
+    ta->newest_stop_durable_ts = WT_TS_NONE;
+    ta->newest_stop_ts = WT_TS_MAX;
+    ta->newest_stop_txn = WT_TXN_MAX;
+    ta->oldest_start_ts = WT_TS_NONE;
+    ta->oldest_start_txn = WT_TXN_NONE;
+    ta->prepare = false;
+}
+
+/*
+ * __wt_time_aggregate_init_max --
+ *     Initialize the fields in an aggregated time window to maximum values, since this structure is
+ *     generally populated by iterating over a set of timestamps and calculating max/min seen for
+ *     each value, it's useful to be able to start with a negatively initialized structure.
+ */
+static inline void
+__wt_time_aggregate_init_max(WT_TIME_AGGREGATE *ta)
+{
+    /* TODO: It's odd that the durable values here match those of the initialization above */
+    ta->newest_start_durable_ts = WT_TS_NONE;
+    ta->newest_stop_durable_ts = WT_TS_NONE;
+    ta->newest_stop_ts = WT_TS_NONE;
+    ta->newest_stop_txn = WT_TXN_NONE;
+    ta->oldest_start_ts = WT_TS_MAX;
+    ta->oldest_start_txn = WT_TXN_MAX;
+    ta->prepare = false;
```

Comment:
Same here.

Reply:
The "initialize to max" is a really limited function, it's only used by reconciliation, so I'm inclined to leave this as it is. Let me know if you feel strongly.